controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -453,6 +454,21 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(ctx context.Contex
 		}
 	}
 
+	if !internal.HasAnnotationKey(controlplanev1.ScaleDownConfigMapEntryRemovedAnnotation)(machineToDelete) {
+		if err := r.managementCluster.TargetClusterControlPlaneIsHealthy(ctx, util.ObjectKey(cluster), kcp.Name); err != nil {
```

Comment:
Not a blocker, but we need to check provider contracts around load balancers to ensure they stop sending requests to machines.